fairscale/optim/oss.py

```diff
@@ -52,7 +52,7 @@ class OSS(Optimizer):
         group (group):
             torch.distributed group (default: group.WORLD)
         broadcast_buffer_size (int):
-            the size of the buffer used to batch the small parameter tensors (default 128k).
+            the max size of the buffer used to batch the small parameter tensors, in number of elements (default 16M).
```

Comment:
talk a bit about the impact on per-GPU memory usage here?

Reply:
the beauty of this PR is that the impact on GPU usage is now.. 0 (zero) :) the small parameters are just a view of a bigger buffer, so the bucketing takes no extra space (on top of taking no extra cpu cycle).
I just realized that I should resize it to match the last bucketed parameter though, right now the remainder is unused and do increase the gpu memory a tiny bit

Reply:
hmm, the test jobs do show a slightly larger memory consumption, looking into that

Reply:
ok, makes sense given the initial extra memory allocation, before the parameters have been folded to the buffer, I've updated the docstring to reflect that